src/PhpUnit/AssertsTrait.php

```diff
@@ -13,6 +16,11 @@
 trait AssertsTrait
 {
     /**
+     * @var Factory
+     */
+    private $constraintFactory;
```

Comment:
My idea was you should "override" the method `getValidator` with your custom needs instead expose the factory.